host/xtest/regression_1000.c

```diff
@@ -902,6 +904,140 @@ static void xtest_tee_test_1011(ADBG_Case_t *c)
 ADBG_CASE_DEFINE(regression, 1011, xtest_tee_test_1011,
 		"Test TA-to-TA features with User Crypt TA");
 
+struct xtest_session {
+	ADBG_Case_t *c;
+	TEEC_Session session;
+	TEEC_Context context;
+};
+
+static TEEC_Result do_open_session(struct xtest_session *cs,
+				   const TEEC_UUID *uuid)
+{
+	TEEC_Result res = TEEC_ERROR_GENERIC;
+	uint32_t ret_orig = 0;
+
+	TEEC_InitializeContext(NULL, &cs->context);
+
+	res = TEEC_OpenSession(&cs->context,
+			       &cs->session, uuid,
+			       TEEC_LOGIN_PUBLIC, NULL, NULL,
+			       &ret_orig);
+	if (res != TEEC_SUCCESS) {
+		TEEC_FinalizeContext(&cs->context);
+		return res;
+	}
+
+	return res;
+}
+
+static TEEC_Result do_get_counter(struct xtest_session *cs,
+				  uint32_t *counter)
+{
+	TEEC_Result res = TEEC_ERROR_GENERIC;
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	uint32_t ret_orig = 0;
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_VALUE_OUTPUT,
+					 TEEC_NONE,
+					 TEEC_NONE,
+					 TEEC_NONE);
+
+	res = TEEC_InvokeCommand(&cs->session,
+				 TA_SIMS_CMD_GET_COUNTER, &op,
+				 &ret_orig);
+	*counter = op.params[0].value.a;
+
+	return res;
+}
+
+static TEEC_Result do_panic(struct xtest_session *cs)
+{
+	return TEEC_InvokeCommand(&cs->session, TA_SIMS_CMD_PANIC,
+				  NULL, NULL);
+}
+
+static void do_close_session(struct xtest_session *cs)
+{
+	TEEC_CloseSession(&cs->session);
+	TEEC_FinalizeContext(&cs->context);
+}
+
+static int test_ta_panic_behavior(ADBG_Case_t *c, const TEEC_UUID *uuid,
+				  bool multi_instance)
+{
+	TEEC_Result exp_res = TEEC_ERROR_GENERIC;
+	uint32_t counter = 0;
+	uint32_t exp_counter = 0;
+	uint32_t ret = 0;
```

Comment:
default set `ret = 1;`, this defaults to an error report. Line 1030 will validate successful completion with `ret = 0;`.

Reply:
I would have wanted initially to completely remove the return state from `test_ta_panic_behavior()` since we are not interpreting at all the output of the test execution, all logs being inside the function already.